src/base-element.js

```diff
@@ -586,40 +597,78 @@ export class BaseElement {
    * The handler is only invoked by events with trust equal to or greater than
    * `minTrust`. Otherwise, a user error is logged.
    *
-   * @param {string} method
+   * @param {string} alias
    * @param {function(!./service/action-impl.ActionInvocation)} handler
    * @param {ActionTrust} minTrust
    * @public
    */
-  registerAction(method, handler, minTrust = ActionTrust.HIGH) {
+  registerAction(alias, handler, minTrust = ActionTrust.HIGH) {
+    this.registerAction_(
+        alias, handler, minTrust, /* opt_isDefault */ false);
+  }
+
+  /**
+   * Registers the default action for this component.
+   * @param {function(!./service/action-impl.ActionInvocation)} handler
+   * @param {string=} alias
+   * @param {ActionTrust} minTrust
+   */
+  registerDefaultAction(
+    handler,
+    alias = DEFAULT_METHOD,
+    minTrust = ActionTrust.HIGH) {
+    this.registerAction_(
+        alias, handler, minTrust, /* opt_isDefault */ true);
+  }
+
+  /**
+   * @param {string} alias
+   * @param {function(!./service/action-impl.ActionInvocation)} handler
+   * @param {ActionTrust} minTrust
+   * @param {boolean} isDefault Whether this action is the default.
+   * @private
+   */
+  registerAction_(alias, handler, minTrust = ActionTrust.HIGH,
+    isDefault = false) {
     this.initActionMap_();
-    this.actionMap_[method] = {handler, minTrust};
+    this.actionMap_[alias] = {handler, minTrust};
+    if (isDefault) {
+      dev().assert(!this.defaultActionAlias_,
+          'There is already a default action registered.');
+      this.defaultActionAlias_ = alias;
+    }
   }
 
   /**
    * Requests the element to execute the specified method. If method must have
    * been previously registered using {@link registerAction}, otherwise an
    * error is thrown.
    * @param {!./service/action-impl.ActionInvocation} invocation The invocation data.
-   * @param {boolean} unusedDeferred Whether the invocation has had to wait any time
+   * @param {boolean=} unusedDeferred Whether the invocation has had to wait any time
    *   for the element to be resolved, upgraded and built.
    * @final
    * @package
    */
   executeAction(invocation, unusedDeferred) {
-    if (invocation.method == 'activate') {
-      if (invocation.satisfiesTrust(this.activationTrust())) {
-        return this.activate(invocation);
-      }
+    const {method} = invocation;
+    const isDefaultMethod = (method == DEFAULT_METHOD);
```

Comment:
Is comparing against `DEFAULT_METHOD` here necessary given that `registerDefaultAction()` already uses it as a default parameter?

Reply:
Ah, I had imagined `registerDefaultAction()` slightly differently. In that case, we can simplify it like this:

```js
let {method} = invocation;
// If the default action has an alias, the handler will be stored under it.
if (method === DEFAULT_METHOD) {
  method = this.defaultActionAlias_ || method;
}
this.initActionMap_();
const holder = this.actionMap_[method];
user().assert(holder, `Method not found: ${method} in %s`, this);
const {handler, minTrust} = holder;
if (invocation.satisfiesTrust(minTrust)) {
  return handler(invocation);
}
```

Though I think it's slightly more clear if we flip it and always store handlers for default action under `DEFAULT_ACTION`. Either way though.